From c5b703e7c1a3f5da43a3864536cfa606d112d60a Mon Sep 17 00:00:00 2001
From: David Faure <faure@kde.org>
Date: Fri, 17 Oct 2014 21:19:36 +0200
Subject: [PATCH] Fix crash on logout.

Turns out the check for kapp->sessionSaving inside queryExit was confusing,
it's actually false when aboutToQuit() leads to queryExit, even during logout.
So always delete the part in slotOnShutdown, called by aboutToQuit().

BUG: 336417
FIXED-IN: 4.14.3
---
 akregator/src/mainwidget.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/akregator/src/mainwidget.cpp b/akregator/src/mainwidget.cpp
index 3211a50..14cd2a9 100644
--- a/akregator/src/mainwidget.cpp
+++ b/akregator/src/mainwidget.cpp
@@ -317,6 +317,8 @@ void Akregator::MainWidget::slotOnShutdown()
 {
     m_shuttingDown = true;
 
+    delete m_part;
+
     // close all pageviewers in a controlled way
     // fixes bug 91660, at least when no part loading data
     while ( m_tabWidget->count() > 1 ) { // remove frames until only the main frame remains
@@ -333,6 +335,7 @@ void Akregator::MainWidget::slotOnShutdown()
 
     delete m_mainTab;
     delete m_mainFrame;
+    m_mainFrame = 0;
 
     Settings::self()->writeConfig();
 }
-- 
1.8.4.1

